todo_app/data/trello_items.py

```diff
@@ -0,0 +1,73 @@
+import os
+from datetime import datetime
+from typing import Dict, List, Optional
+
+import requests
+
+from todo_app.data.items import Item, Status
+
+
+class TrelloRequests:
+    def __init__(self):
+        self.errored = False
+        self._board_id: str = os.getenv("TRELLO_BOARD_ID")
+        self._params: Dict[str, str] = {"key": os.getenv("TRELLO_KEY"), "token": os.getenv("TRELLO_TOKEN")}
+        self._url = f"https://api.trello.com/1"
+        self._lists_ids_to_names = None
+        self._names_to_list_ids = None
+
+    def init_list_maps(self) -> None:
+        if self._lists_ids_to_names is None:
+            self._lists_ids_to_names = self._get_lists()
+            if self.errored:
+                return
+            self._names_to_list_ids = {v: k for k, v in self._lists_ids_to_names.items()}
+
+    def _get_lists(self) -> Dict[str, Status]:
+        r = self._get(f"{self._url}/boards/{self._board_id}/lists", self._params)
+        return {l["id"]: Status(l["name"]) for l in r.json()} if r is not None else None
+
+    def _json_to_item(self, json: Dict[str, str]) -> Item:
+        return Item(
+            id_=json["id"],
+            status=self._lists_ids_to_names[json["idList"]],
+            title=json["name"],
+            description=json["desc"],
+            due=datetime.strptime(json["due"][:10], "%Y-%m-%d") if json["due"] else None,
+        )
+
+    def _get(self, url: str, params: Dict[str, str]) -> requests.Response:
+        r = requests.get(url, params=params)
+        self._set_errored(r)
+        return r if not self.errored else None
+
+    def _set_errored(self, r: requests.Response) -> None:
+        if r.status_code != 200:
+            self.errored = True
```

Comment:
This is fine as long as Trello don't decide to start returning another 2XX code at any point; we could instead switch to use the `ok` property on responses that will check that a little more carefully:
```suggestion
        if !r.ok:
            self.errored = True
```

Reply:
ooo I did not know about this